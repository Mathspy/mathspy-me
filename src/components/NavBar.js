import React from "react";

import { css } from "emotion";

const NavIcon = ({ children }) => (
  <div className={containerStyle}>{children}</div>
);

const containerStyle = css`
  display: flex;
  background-color: #3d1755;
  align-items: center;
  justify-content: space-between;
  padding: 11px;

  @media (min-width: 600px) {
    justify-content: flex-start;
    padding: 10px;
    flex-direction: column;
  }

  @media (min-width: 900px) {
    padding: 17px;
  }
`;

export default NavIcon;
