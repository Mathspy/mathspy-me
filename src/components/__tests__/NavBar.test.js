import React from "react";

import { shallow } from "enzyme";

import NavBar from "../NavBar";

describe("NavBar", () => {
  it("should be a simple div", () => {
    const wrapper = shallow(<NavBar />);

    expect(wrapper.is("div")).toBe(true);
  });

  it("should render the children passed to it", () => {
    const wrapper = shallow(
      <NavBar>
        <div className="test" />
      </NavBar>
    );

    expect(wrapper.contains(<div className="test" />)).toBe(true);
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<NavBar />);

    expect(wrapper).toMatchSnapshot();
  });
});
