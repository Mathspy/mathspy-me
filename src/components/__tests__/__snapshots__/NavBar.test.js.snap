// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #3d1755;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 11px;
}

@media (min-width:600px) {
  .emotion-0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 10px;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:900px) {
  .emotion-0 {
    padding: 17px;
  }
}

<div
  className="emotion-0"
/>
`;
